patches/minecraft/net/minecraft/item/ItemArmor.java.patch

```diff
@@ -41,8 +55,30 @@
 +                case GOLD:    return Items.field_151043_k;
 +                case IRON:    return Items.field_151042_j;
 +                case DIAMOND: return Items.field_151045_i;
-+                default:      return customCraftingMaterial;
++                default:      return Items.field_190931_a;
```

Comment:
This whole patch for `func_151685_b` should be set to the way it was in vanilla, and then anything calling it should check for `null` return value instead of `Items.AIR`.